src/main/scala/RockPaperScissors.scala

```diff
@@ -0,0 +1,188 @@
+    case (input) if(input == "paper") =>  paperWeight += 1
+    case (input) if(input == "scissor") => scissorsWeight += 1
+  }
+
+
+  def determineWinner(moveAI: String, movePlayer: String): String = (moveAI) match
+  {
+    case(a) if(moveAI == movePlayer) => "draw against first A.I"
+    case(a) if(moveAI == "paper" && movePlayer == "rock") => winning -= 1; "lose against first A.I"
+    case(a) if(moveAI == "rock" && movePlayer == "scissor") => winning -= 1; "lose against first A.I"
+    case(a) if(moveAI == "scissor" && movePlayer == "paper") => winning -= 1;  "lose against first A.I"
+    case(a) if(moveAI == "rock" && movePlayer == "paper" ) => winning += 1; "win against first A.I"
+    case(a) if(moveAI == "scissor" && movePlayer =="rock" ) => winning += 1; "win against first A.I"
+    case(a) if(moveAI == "paper"&& movePlayer == "scissor") =>  winning += 1; "win against first A.I"
+
+  }
+
+  def determineOveralWinner(): Unit =
+  {
+     if(winning > 0) println("first player has won by " + winning)
+     else if (winning < 0) println("first player has lost by " + winning)
+     else println("neither player lost or won")
+  }
+
+  def runAIVSAI(totalGames: Int) =
+  {
+    var totalGamesInFunction = totalGames
+    while (totalGamesInFunction > 0)
+    {
+      var moveAI = commitAImove(generateAIMove(), scissorsWeight, rockWeight, paperWeight ,weightGroupsSizeO )
+      var movePlayer = commitAImove(generateAIMove(), scissorsWeight, rockWeight, paperWeight, weightGroupsSizeO )
+      weightAdjustmentCheck(movePlayer)
+      println(determineWinner(moveAI, movePlayer ))
+      println("genrated Ai move " + moveAI)
+      totalGamesInFunction -= 1
+    }
+    determineOveralWinner()
+  }
+
+  def runAIVSPlayer (totalGames: Int) =
+  {
+    var totalGamesInFunction = totalGames
+    while (totalGamesInFunction > 0)
+    {
+      var moveAI = commitAImove(generateAIMove(), scissorsWeight, rockWeight, paperWeight , weightGroupsSizeO)
+      var movePlayer = takeInput()
+      weightAdjustmentCheck(movePlayer)
+      println(determineWinner(moveAI, movePlayer))
+      println("genrated Ai move " + moveAI)
+      totalGamesInFunction -= 1
+    }
+    determineOveralWinner()
+  }
+
+
+  //functions for multiple AI players
+  def determinWinnerMulipleAI(allCommitedMoves: Map[Int, String]): Map[Int, String] =
+  {
+      var currentAllCommitedMoves: Map[Int , String]  = allCommitedMoves
+      currentAllCommitedMoves.values.foreach(item => println("generated Items" + item))
+      var scissorCount = 0
+      var rockCount = 0
+      var paperCount = 0
+      allCommitedMoves.values.foreach(item =>
+        if(item  == "scissor") scissorCount += 1
+        else if (item == "rock") rockCount += 1
+        else if(item == "paper") paperCount += 1  )
+
+      (scissorCount) match  //move to own function when everything is 100% working
+      {
+        case (scissor) if(paperCount == 0 && scissorCount > 0 && rockCount > 0
```

Comment:
the three cases seem to be almost or exactly the same, could you not make it so that they're a method which you would call on a particular case and deal with some parameters being passed or something along those lines